generator/src/main/java/com/scottlogic/deg/generator/generation/string/RegexStringGenerator.java

```diff
@@ -408,28 +421,42 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(this.automaton, this.getClass());
     }
+    /** throws @IllegalArgumentException if a or b less than zero */
+    private static boolean multiplyWouldOverflow(long a, long b) {
+        Validate.isTrue(a >= 0, "The value must be >=0: %d", a);
+        Validate.isTrue(b >= 0, "The value must be >=0: %d", b);
+        return a > 0 && b > 0 && Long.MAX_VALUE / b < a;
+    }
 
-    public static class UnionCollector {
+    /** throws @IllegalArgumentException if a or b less than zero */
+    private static boolean addWouldOverflow(long a, long b) {
+        Validate.isTrue(a >= 0, "The value must be >=0: %d", a);
+        Validate.isTrue(b >= 0, "The value must be >=0: %d", b);
+        return a > 0 && b > 0 && (Long.MAX_VALUE - b < a);
```

Comment:
is it really a worry that a or be could be 0 or less.

i think these methods would still be safe if they were just the ```long.max - b < a```  (maybe check for 0 in multiply to avoid divide by zero)

Reply:
It does not make sense for the class t o be using negatives for the values its passing in and i don't want to complicate my new methods to handle them.  But i get weird behaviour if negative passed in as if b is negative we can overflow in our expression (MAX - b < a) using to check if can get an overflow.  Putting in the assert made me find another bug in code i did so at moment i'd like to leave it there as it will alert us if our assumption is broken

Reply:
regarding moving closer to where called, they are called in 2 different inner classes so needs to go in common place that both see.  We also need to decide what our code style is on that front as the approach you suggest is not necessarily one everyone follows at moment.  We should probably agree a consistent style regarding ordering of methods.